packages/astro/src/core/ssr/resolve-config-paths.ts

```diff
@@ -0,0 +1,94 @@
+import { tsconfigResolverSync, TsConfigResultSuccess } from 'tsconfig-resolver';
+import fs from 'fs'
+import { posix as ps } from 'path'
+import type * as vite from 'vite'
+
+/** Return the nearest tsconfig.json or jsconfig.json configuration result. */
+const getConfigResult = (): TsConfigResultSuccess | null => {
+	/** Closest tsconfig.json */
+	const tsConfig = tsconfigResolverSync({ searchName: 'tsconfig.json' })
+
+	// return tsconfig.json, if it exists
+	if (tsConfig.exists) return tsConfig
+
+	/** Closest jsconfig.json */
+	const jsConfig = tsconfigResolverSync({ searchName: 'jsconfig.json' })
+
+	// return jsconfig.json, if it exists
+	if (jsConfig.exists) return jsConfig
+
+	// return null if no config exists
+	return null
+}
+
+/** Return the resolvers from the tsconfig.json or jsconfig.json configuration. */
+const getConfigResolver = (): { baseUrl: string, paths: { [alias: string]: string[] } } | null => {
+	/** Configuration result */
+	const configResult = getConfigResult()
+
+	if (configResult && configResult.config.compilerOptions?.baseUrl) {
+		let { baseUrl, paths } = configResult.config.compilerOptions as {
+			baseUrl: string
+			paths: {
+				[key: string]: string[];
+			}
+		}
+
+		// resolve the base url from the configuration file directory
+		baseUrl = ps.resolve(ps.dirname(configResult.path), baseUrl)
+
+		// guarantee paths is an object
+		paths = Object(paths)
```

Comment:
Is this just getting TypeScript to play nice or is it actually necessary?

Reply:
This is necessary, because `paths` comes from the user config. However, I have refactored this in 886ef899d62cc991c9925e29240b5ebae7ea263b to happen within the area of code labeled "sanitize".